### PR TITLE
feat(py): support Veo image-to-video via GenerateVideosSource

### LINKED ISSUE #4363: [Python] Support image input in Veo for image-to-video
## Description

Support image input if Veo supports it for image-to-video use cases.

## Location
- File: `py/plugins/google-genai/src/genkit/plugins/google_genai/models/veo.py`
- Line: 282
- Current code: `# TODO: Support image input if Veo supports it (e.g. for image-to-video)`

## Acceptance Criteria
- [ ] Check if Veo API supports image input
- [ ] Implement image-to-video functionality if supported

### PR DESCRIPTION
**Pull request by a2105z**
## Summary
- Veo API supports image-to-video; wire Genkit `image/*` media parts into `google.genai.types.GenerateVideosSource`
- Extract images from the last message (`data:` base64 and `gs://`), matching JS `extractVeoImage`
- Both background `start` and blocking Vertex `generate` paths now pass `source=` (prompt and/or image)
- Unit tests cover extraction edge cases (video ignored, last-message-only, image-only)

Fixes #4363

## Test plan
- [x] `pytest packages/genkit-google-genai/tests/veo_test.py --no-cov`
- [x] Unit coverage for image extraction (`data:` / `gs://`), last-message selection, video parts ignored, `source=` wiring for start/generate
- [ ] CI green on this PR
- [ ] Manual (optional): Vertex Veo image-to-video with a real image part

### CODE REVIEW

File: py/packages/genkit-google-genai/src/genkit_google_genai/models/veo.py [hunk 1]
```diff
@@ -142,6 +143,69 @@ def _extract_text(request: ModelRequest) -> str:
...
+            if not payload:
+                return None
+            try:
+                image_bytes = base64.b64decode(payload, validate=False)
+            except (ValueError, TypeError):
+                return None
+            return genai_types.Image(image_bytes=image_bytes, mime_type=content_type)
+
+        # http(s) and other URL schemes are not inlined here (JS only accepts data: URLs).
+        return None
```
[the remark this answers is not in the document]
Comment (a2105z) [reply py/packages/genkit-google-genai/src/genkit_google_genai/models/veo.py]: Done — unsupported/invalid image parts now `continue` so later supported parts (e.g. `gs://`) are still found. Added a regression test.

File: py/packages/genkit-google-genai/src/genkit_google_genai/models/veo.py [hunk 2]
```diff
@@ -142,6 +143,69 @@ def _extract_text(request: ModelRequest) -> str:
     return ' '.join(prompt_parts)
 
 
+def _media_content_type(media: Media) -> str:
+    """Resolve a media content type from the part or a data: URL prefix."""
+    if media.content_type:
+        return media.content_type
+    url = media.url or ''
```
[the remark this answers is not in the document]
Comment (a2105z) [reply py/packages/genkit-google-genai/src/genkit_google_genai/models/veo.py]: Done — `_media_content_type` and the extract loop now nil-guard `media`.

**Comment by a2105z** [comment]
Hi @genkit-ai/genkit-googlers @pavelgj @apascal07 @huangjeff5 @ifielker @MichaelDoyle @yesudeep @ssbushi @shrutip90 @cabljac @adesinah — circling back on Python Veo image-to-video support.

It has been open for a couple of weeks and is still waiting on a maintainer review. Would someone be able to take a look when you have a moment? Happy to address any feedback. Thank you!
